Allow the build date/time to be overridden by BUILD_DATE

jmdh opened the pull request:
Including the current date/time in the build means that it cannot be
reproducible. Instead, BUILD_DATE can be set to a value from the source
(in the Debian project, this will be the latest changelog date).

See https://wiki.debian.org/ReproducibleBuilds/TimestampsFromCPPMacros
for more information.

jmdh committed:
Allow the build date/time to be overridden by BUILD_DATE

Including the current date/time in the build means that it cannot be
reproducible. Instead, BUILD_DATE can be set to a value from the source
(in the Debian project, this will be the latest changelog date).

See <https://wiki.debian.org/ReproducibleBuilds/TimestampsFromCPPMacros>
for more information.

miwob commented:
I'm basically ok with your PR, but I currenly see two things that bother me:

1) date '+%b %e %Y'  only returns the current date not including the time.

2) since you're running 'date' from the shell, it possibly returns the month name in a language other than English depending on the systems locale, and therefore the RPL_CREATED numeric reply might look inconsistent compared to other numerics since we use English everywhere.

jmdh commented:
Thanks. I've pushed a new version which sets LANG=c and adds the missing time, which hopefully addresses your concerns. I've also moved the #define to the top of the file, as suggested on IRC.

miwob commented:
Implemented with slightly modifications as discussed on irc.
